api/api.go

```diff
@@ -144,30 +146,48 @@ func App(opts ...options.AppOption) (*fiber.App, error) {
 
 	// Auth middleware checking if API key is valid. If no API key is set, no auth is required.
 	auth := func(c *fiber.Ctx) error {
-		if len(options.ApiKeys) > 0 {
-			authHeader := c.Get("Authorization")
-			if authHeader == "" {
-				return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{"message": "Authorization header missing"})
-			}
-			authHeaderParts := strings.Split(authHeader, " ")
-			if len(authHeaderParts) != 2 || authHeaderParts[0] != "Bearer" {
-				return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{"message": "Invalid Authorization header format"})
-			}
+		if len(options.ApiKeys) == 0 {
+			return c.Next()
+		}
 
-			apiKey := authHeaderParts[1]
-			validApiKey := false
-			for _, key := range options.ApiKeys {
-				if apiKey == key {
-					validApiKey = true
-				}
+		// Check for api_keys.json file
+		fileContent, err := ioutil.ReadFile("api_keys.json")
```

Comment:
to keep it simple we can also just load the file once on start and pass the api keys down the line as a list

Reply:
@mudler needs to be reloaded on request so we dont have to restart the program everytime is the goal, this way we can add keys as its running

Reply:
> @lunamidori5 - Do we know what the performance impact of `ioutil.ReadFile` on every request will be? @mudler is more familiar with golang and would know if this is a concern or not - it might be worth some performance testing in the meantime.
> 
> Personally, I would be inclined to use something like https://pkg.go.dev/github.com/fsnotify/fsnotify to watch the file and keep options.ApiKeys up-to-date - that way, we aren't hitting the filesystem on every single request?

I guess? I dont know much about GO...